docs/application/native/guides/internationalization/i18n.md

```diff
@@ -773,6 +777,102 @@ The `current()` and `next()` functions only check the current index against the
+4. To obtain formatted message, you need to use one of i18n_plural_format_format_*, which also takes
+a number object and returns corresponding message based on the defined rules.
+
+
+### Pattern interpretation:
+
+The pattern text defines the message output for each plural case of the specified locale.  
+**Syntax:**
+
+```
+pluralStyle = [offsetValue] (selector '{' message '}')+
+offsetValue = "offset:" number
+selector = explicitValue | keyword
+explicitValue = '=' number  // adjacent, no white space in between
+keyword = [^[[:Pattern_Syntax:][:Pattern_White_Space:]]]+
+message: message = messageText (argument messageText)*
+argument = noneArg | simpleArg | complexArg
+complexArg = pluralArg | selectordinalArg
+noneArg = '{' argNameOrNumber '}'
+simpleArg = '{' argNameOrNumber ',' argType [',' argStyle] '}'
+pluralArg = '{' argNameOrNumber ',' "plural" ',' pluralStyle '}'
+selectordinalArg = '{' argNameOrNumber ',' "selectordinal" ',' pluralStyle '}'
+argNameOrNumber = argName | argNumber
+argName = [^[[:Pattern_Syntax:][:Pattern_White_Space:]]]+
+argNumber = '0' | ('1'..'9' ('0'..'9')*)
+argType = "number" | "date" | "time" | "spellout" | "ordinal" | "duration"
+argStyle = "short" | "medium" | "long" | "full" | "integer" | "currency" | "percent" | argStyleText
+```
+
+**Examples:**
+
+```
+one{dog} other{dogs}                 // English plural format for dog
+one{pies} few{psy} other{psów}       //Polish translation of dog in plural format
+```
+
+Pattern_White_Space between syntax elements is ignored, except between the {curly braces}
+and their sub-message, and between the '=' and the number of an explicitValue.
+
+In International Components for Unicode (ICU) there are six predefined keywords:
+`zero`, `one`, `two`, `few`, `many` and `other`. It is obligatory to define message text for the `other` keyword,
+which is included in every rule set. In case if you do not define the specified message for a particular number object,
+the `other` keyword message will be returned to this plural case.
+
+When formatting, the input number is first matched against the explicitValue clauses.
+If there is no exact-number match, then a keyword is selected by calling the Plural Rules
+with the input number minus the offset. (The offset defaults to 0 if it is omitted from the pattern string.)
+If there is no clause with that keyword, then the "other" clauses is returned.
+
+An unquoted pound signal (#) is a selected sub message (that is, outside of argument nested in the sub message)
+is replaced by the input number-minus-offset value. The `number-minus-offset` value is formatted
+using a Number format for the PluralFormat locale.
+
+> **Note:**  
+> That argument is formatting without subtracting the offset.
+> If you need a custom format and have a non-zero-offset, then you need to pass the `number-minus-offset` value as a separate parameter.
+
+### Defining Custom Plural Rules:
```

Comment:
Please change as:
Define Defining Custom Plural Rules

Reply:
I assume, you mean to change to: Define Custom Plural Rules

Reply:
@lu-pik , Yes please change it to "Define Custom Plural Rules"